crates/typst-library/src/text/raw.rs

```diff
@@ -126,6 +132,49 @@ pub struct RawElem {
     /// ````
     #[default(HorizontalAlign(GenAlign::Start))]
     pub align: HorizontalAlign,
+
+    /// A set of additional syntax definitions to load.
+    /// The syntax definitions should be in the `sublime-syntax` file format.
+    /// Syntaxes are a dictionary of the language name and the syntax file.
+    ///
+    /// ````example
+    /// #set raw(syntaxes: "SExpressions.sublime-syntax")
+    ///
+    /// ```sexp
+    /// (defun factorial (x)
+    ///    (if (zerop x)
+    ///        ; with a comment
+    ///        1
+    ///        (* x (factorial (- x 1)))))
+    /// ```
+    /// ````
+    #[parse(
+        let (syntaxes, data) = if let Some(Spanned { v: paths, span }) = args.named::<Spanned<SyntaxPaths>>("syntaxes")? {
+            // Load bibliography files.
+            let data = paths.0
+                .iter()
+                .map(|path| {
+                    let id = vm.location().join(path).at(span)?;
+                    vm.world().file(id).at(span)
+                })
+                .collect::<SourceResult<Vec<Bytes>>>()?;
+
+            // Check that parsing works.
+            let _ = load(&paths, &data).at(span)?;
+
+            (Some(paths), Some(data))
+        } else {
+            (None, None)
+        };
+
+        syntaxes
+    )]
+    pub syntaxes: SyntaxPaths,
```

Comment:
this should probably have `#[fold]` and implement the `Fold` trait so that two separate set rules combine their syntaxes instead of one overriding the other one.